### DISCUSSION TITLE
Can I do conditional includes?

### DISCUSSION THREAD

**Discussion by operdeck** [quarto-dev/quarto-cli#11564] [Q&A] [answered]
### Description

I have a quarto notebook that basically works off a set of three datasets, with each one being optional. The document consists of a number of plots specific to each of these datasets. If a dataset is missing, I currently have an explicit test around each plot and instead present a text like "not available".

This isn't very elegant - there's a ton of these "not available" texts, the normal text blocks in between still show, and it's a lot of repetitive code. 

I came across the possibility to include separate markdown files: {{< include _content.qmd >}}. I could split up the current document into three, but what I'm really missing is an option to _conditionally_ include these. Is there a way to accomplish this?

Thanks!

**Comment by mcanouil** [comment, marked as answer]
Note that this particular topic has been discussed several times, [search for "conditional"](https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=conditional).

- https://github.com/gadenbuie/quarto-partials
- use computation to handle the conditional part
- use metadata, etc.: https://quarto.org/docs/authoring/conditional.html

**Comment by operdeck** [comment]
Thank you Mickaël! The metadata check is what I was looking for: {.content-hidden unless-meta="path.to.metadata"}. That should cut it.
